subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/BasicJavaCompilerIntegrationSpec.groovy

```diff
@@ -252,91 +252,171 @@ class Main {
         file('build/classes/main/Java$$Generated.java').exists()
     }
 
-    def writeAnnotationProcessorProject() {
-        file("processor").create {
+    @LeaksFileHandles("holds processor.jar open for in process compiler")
+    def "can use annotation processor in compileOnly configuration"() {
+        when:
+        buildFile << """
+            apply plugin: "java"
+            dependencies {
+                compileOnly project(":processor")
+            }
+        """
+        settingsFile << "include 'annotation', 'processor'"
+        writeAnnotationProcessorProjects("annotation", "processor", "SimpleAnnotation", "Generated")
+
+        file("src/main/java/Java.java") << "@com.test.SimpleAnnotation public class Java {}"
+
+        then:
+        succeeds("compileJava")
+        file('build/classes/main/Java$$Generated.java').exists()
+    }
+
+    @LeaksFileHandles("holds processor.jar open for in process compiler")
```

Comment:
I have no idea, this is a "cargo-cult copy-paste" from the existing `"can use annotation processor"` test.